presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentStatusUtils.java

```diff
@@ -0,0 +1,146 @@
+    }
+
+    public static void stopPresto(StopType stopType)
+            throws PrestoManagerException
+    {
+        if (executeCommand("service presto status") != 0) {
+            throw new PrestoManagerException("Presto is not running");
+        }
+        switch (stopType) {
+            case TERMINATE:
+                if (executeCommand("service presto stop") != 0) {
+                    throw new PrestoManagerException("Failed to stop presto");
+                }
+                break;
+            case KILL:
+                executeCommand(format("sudo %s kill", LAUNCHER_SCRIPT));
+                if (executeCommand("service presto status") != 3) {
+                    throw new PrestoManagerException("Failed to stop presto");
+                }
+                break;
+            case GRACEFUL:
+                try {
+                    if (getFileProperty(PATH_TO_CONFIG_FILE, "coordinator").equalsIgnoreCase("true")) {
+                        throw new PrestoManagerException("Coordinator can't be gracefully stopped");
+                    }
+                    String prestoPort = getFileProperty(PATH_TO_CONFIG_FILE, "http-server.http.port");
+                    Response response = newRequestBuilder(new URI(format("localhost:%s/v1/info/state", prestoPort)))
+                            .withBody(json("SHUTTING_DOWN")).buildPut().invoke();
+                    if (response.getStatus() != 200) {
+                        throw new PrestoManagerException("Failed to stop presto gracefully");
+                    }
+                }
+                catch (IOException | URISyntaxException e) {
+                    throw new PrestoManagerException("Error in stopping presto", e);
+                }
+        }
+    }
+
+    public static void restartPresto()
+            throws PrestoManagerException
+    {
+        if (executeCommand("rpm -q presto-server-rpm") != 0) {
+            throw new PrestoManagerException(format("Presto not installed"));
+        }
+        if (executeCommand("service presto restart") != 0) {
+            throw new PrestoManagerException("Failed to restart presto");
+        }
+    }
+
+    public static Response prestoStatus()
+            throws PrestoManagerException
+    {
+        if (executeCommand("rpm -q presto-server-rpm") != 0) {
+            return Response.status(OK).entity(json("Presto not installed")).build();
+        }
+        if (executeCommand("service presto status") != 0) {
+            return Response.status(OK).entity(json("Presto is installed but not running")).build();
+        }
+        try {
+            String prestoPort = getFileProperty(PATH_TO_CONFIG_FILE, "http-server.http.port");
+            return newRequestBuilder(new URI(format("localhost:%s/v1/info", prestoPort))).buildGet().invoke();
+        }
+        catch (IOException | URISyntaxException e) {
+            throw new PrestoManagerException("Failed to get status");
+        }
+    }
+
+    public enum StopType
```

Comment:
This should be replaced with a `JaxrsParameter` (as seen in the logs API pull request), or with a `ParamConverter`/`ParamConverterProvider` once we figure out how to use those.